kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerConfiguration.java

```diff
@@ -79,13 +79,22 @@ public String getContainerName() {
         return this.containerName;
     }
 
+    /**
+     * Returns the list of mapped ports that will be mapped to the given container
+     *
+     * @return
+     */
+    public List<ContainerPort> getContainerPorts() {
+        return this.containerPorts;
+    }
+
     /**
      * Returns the list of external ports that will be mapped to the given container
      *
      * @return
      */
     public List<Integer> getContainerPortsExternal() {
```

Comment:
Maybe we can consider deprecating (without removing) the API methods that allow to get/set the port mappings as a list of integer suggesting to use the new `ContainerPort` APIs, since the former provide incomplete information (you don't know the protocol).

The deprecation should be done in a follow-up PR to adhere with conventional commits.

Reply:
Resolving this for now, but have added to my to-do list.